### DISCUSSION TITLE
scaled_iota name

### DISCUSSION THREAD

**Discussion by jtlap** [jfalcou/eve#1268] [Done & Done!]
iota is a good name as it is the same as in std, but scaled_iota is not.
When I read scaled_iota I think of iota(...)*scale and that's not it.

proposals
 arithmetic_progression : a bit long
.ap : a bit short
arith_prog  or arithmetic_prog are midway

progression can also be replaced by sequence or seq.
It is also possible to write new functions on the same model geom_seq harmonic_seq...

I thingk I prefer arithmetic_seq
What do you think ?

**Comment by DenisYaroshevskiy** [comment]
It is not `iota(...)*scale` ??

I thought it is.

**Reply by DenisYaroshevskiy** [reply]
Well - the base can be adjusted - but that's also the case with the regular iota.

**Reply by jtlap** [reply]
the code is ok it's the name that is not scaled _iota is what you mean only if the first term is 0.
it computes base+scale*(n-1) which is general arithmetic sequence and scale  is not a "scale" but a "step"

**Reply by jtlap** [reply]
in term of algorithm it's exclusive_scan vs inclusive_scan in a peculiar case

**Reply by DenisYaroshevskiy** [reply]
the C++ standard iota and standard iota view allow for a different initial value

https://en.cppreference.com/w/cpp/ranges/iota_view
https://en.cppreference.com/w/cpp/algorithm/iota

I think it should be called smth `iota` because the user should be able to find it.
If we call it not iota, user will have troble.

**Reply by jfalcou** [reply]
I would prefer `iota_step` as I remember being confused by the scaled suffix when looking at the code and not seeing any scaling happening.

**Comment by DenisYaroshevskiy** [comment]
Ok - I see it confused ppl. Hmm.

The word iota should be and should come first. `iota_step` - eugh.

`iota_with_step`? Maybe?

**Reply by jfalcou** [reply]
iota_with_step soudns OK to me

**Reply by jtlap** [reply]
for me too.
Who does the job ?
I can if you want

**Reply by jtlap** [reply]
perhaps tomorrow...
